contrib/devtools/copyright_header.py

```diff
@@ -0,0 +1,610 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import re
+import fnmatch
+import sys
+import subprocess
+import datetime
+import os
+
+################################################################################
+# file filtering
+################################################################################
+
+EXCLUDE = [
+    # libsecp256k1:
+    'src/secp256k1/include/secp256k1.h',
+    'src/secp256k1/include/secp256k1_ecdh.h',
+    'src/secp256k1/include/secp256k1_recovery.h',
+    'src/secp256k1/include/secp256k1_schnorr.h',
+    'src/secp256k1/src/java/org_bitcoin_NativeSecp256k1.c',
+    'src/secp256k1/src/java/org_bitcoin_NativeSecp256k1.h',
+    'src/secp256k1/src/java/org_bitcoin_Secp256k1Context.c',
+    'src/secp256k1/src/java/org_bitcoin_Secp256k1Context.h',
+    # auto generated:
+    'src/univalue/lib/univalue_escapes.h',
+    'src/qt/bitcoinstrings.cpp',
+    'src/chainparamsseeds.h',
+    # other external copyrights:
+    'src/tinyformat.h',
+    'src/leveldb/util/env_win.cc',
+    'src/crypto/ctaes/bench.c',
+    'qa/rpc-tests/test_framework/bignum.py',
+    # python init:
+    '*__init__.py',
+]
+EXCLUDE_COMPILED = re.compile('|'.join([fnmatch.translate(m) for m in EXCLUDE]))
+
+INCLUDE = ['*.h', '*.cpp', '*.cc', '*.c', '*.py']
```

Comment:
Probably should include *.sh, *.am, *.m4, and *.include

Might make more sense as an exclusion list...